### PR TITLE
Reuse BytesBuffer in BinlogParser

### PR DESCRIPTION
**Pull request by D3Hunter**
to avoid reallocation of buffer every time

although, in our bench test, reusing buf don't contribute much, about 1% CPU(by gc)

### CODE REVIEW

**Comment by D3Hunter** [comment]
@lance6716 PTAL

File: replication/parser.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"bytes"
 	"encoding/binary"
 	"fmt"
+	"github.com/go-mysql-org/go-mysql/utils"
```
Comment (atercattus) [inline replication/parser.go:7]: Use goimports for better imports sorting.

File: replication/parser.go [hunk 2]
```diff
@@ -126,14 +130,15 @@ func (p *BinlogParser) parseSingleEvent(r io.Reader, onEvent OnEventFunc) (bool,
...
+	if n, err = io.CopyN(buf, r, int64(h.EventSize-EventHeaderSize)); err != nil {
 		return false, errors.Errorf("get event err %v, need %d but got %d", err, h.EventSize, n)
 	}
 	if buf.Len() != int(h.EventSize) {
 		return false, errors.Errorf("invalid raw data size in event %s, need %d but got %d", h.EventType, h.EventSize, buf.Len())
 	}
 
-	rawData := buf.Bytes()
+	var rawData []byte
+	rawData = append(rawData, buf.Bytes()...)
```
Comment (atercattus) [inline replication/parser.go:142]: But we still have an allocation of almost the same size here. Is it enough for you?

Comment (D3Hunter) [reply replication/parser.go:142]: syncer of pingcap/dm uses this rawdata, so cannot be removed

**Review by atercattus — APPROVED** [review]

File: replication/parser.go (hunk 2 above)
Comment (D3Hunter) [reply replication/parser.go:142]: in our bench test, reusing buf don't contribute much, about 1% CPU(by gc)

**Review by lance6716 — APPROVED** [review]

**Pull request merged**